src/DurableTask.Netherite/OrchestrationService/NetheriteConfigurationException.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace DurableTask.Netherite
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+
+    /// <summary>
+    /// An exception that indicates configuration errors for Netherite.
+    /// </summary>
+    [Serializable()]
+    public class NetheriteConfigurationException : System.Exception
```

Comment:
As mentioned, the options pattern does have a validation exception type in it: https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.options.optionsvalidationexception?view=dotnet-plat-ext-6.0

With some refactoring, could leverage the options pattern more directly to avoid having to recreate this ourselves.

Reply:
I will leave the refactoring of the settings into an options pattern up to a future PR (perhaps for a major version change). It is not a goal of this PR to align Netherite with best DI practices.